src/Sylius/Bundle/OrderBundle/NumberGenerator/SequentialOrderNumberGenerator.php

```diff
@@ -64,11 +76,30 @@ public function __construct(
      */
     public function generate(OrderInterface $order)
     {
-        $sequence = $this->getSequence();
-        
-        $number = $this->generateNumber($sequence->getIndex());
-        $sequence->incrementIndex();
-        
+        $number = null;
+        $counter = 0;
+        $done = false;
+        while (!$done) {
```

Comment:
We should extract it to separate method. And than:
```php
        for ($i = 0; $i < 100; $i++) {
//            Your stuff 
//            If succeed, early return
        }

        return 0;
```
No extra variable needed. (despite `$i`)